apps/src/util/i18nStringTrackerWorker.js

```diff
@@ -0,0 +1,104 @@
+import $ from 'jquery';
+
+/**
+ * Gets the singleton instance of an I18nStringTrackerWorker
+ * @returns {I18nStringTrackerWorker}
+ */
+export function getI18nStringTrackerWorker() {
+  return new I18nStringTrackerWorker();
+}
+
+/**
+ * A singleton class which buffers i18n string usage data and sends it in batches to the '/i18n/track_string_usage' API.
+ */
+class I18nStringTrackerWorker {
+  constructor() {
+    // Check if there is already a singleton instance.
+    const instance = this.constructor.instance;
+    if (instance) {
+      return instance;
+    }
+
+    // Set the singleton instance to this instance if it doesn't exist already.
+    this.constructor.instance = this;
+
+    /**
+     * A buffer of records to be sent in batches. It should have the following structure:
+     * {
+     *   'http://studio.code.org/s/20-hour/stage/2/puzzle/4': {
```

Comment:
is there any situation in which we'd end up having multiple urls here?

Reply:
No. I have removed the multiple URL feature.